nixos/modules/services/mail/roundcube.nix

```diff
@@ -0,0 +1,89 @@
+{ lib, config, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.roundcube;
+in
+{
+  options.services.roundcube = {
+    enable = mkEnableOption "Roundcube";
+
+    listenAddress = mkOption {
+      type = types.str;
+      default = "[::]";
+      description = "Listening address. IPv6 addresses must be enclosed in square brackets";
+    };
+
+    listenPort = mkOption {
+      type = types.int;
+      default = 80;
+      description = "Listening port";
+    };
+    
+    subDomain = mkOption {
+      type = types.str;
+      example = "webmail";
+      description = "Sub-domain to use which is the name of the nginx vhost";
+    };
+    
+    extraConfig = mkOption {
+      type = types.str;
+      default = ''
+        <?php
+
+        $config = array();
+        $config['db_dsnw'] = 'pgsql://roundcube:pass@localhost/roundcubemail';
```

Comment:
I don't think we should have an hard-coded host + hard-coded password here. This should be make configurable and defaulting to unix sockets also makes password unnecessary:  `pgsql://roundcube@unix(/tmp)/roundcubemail` (this is untested and adapted from: https://serverfault.com/questions/215471/roundcube-cant-connect-to-postgresql-database

Reply:
Ideally the database would be also created automatically.

Reply:
Yes, I would have like to do this later but I can think about it

Reply:
But Can I have help on the test file ?